File: src/main/java/org/springframework/samples/petclinic/rest/dto/PetCreateDto.java
package org.springframework.samples.petclinic.rest.dto;

import java.time.LocalDate;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record PetCreateDto(@NotBlank String name, @NotNull LocalDate birthDate, @NotNull Integer typeId) {
}

File: src/main/kotlin/org/springframework/samples/petclinic/rest/dto/PetCreateDto.kt
package org.springframework.samples.petclinic.rest.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import java.time.LocalDate

data class PetCreateDto(
    @field:NotBlank val name: String,
    @field:NotNull val birthDate: LocalDate,
    @field:NotNull val typeId: Int
)